MovieJournalMVC/Model/Store.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  Store.swift
+//  MovieJournalMVC
+//
+//  Created by Lauren Small on 3/12/19.
+//  Copyright © 2019 Lauren Small. All rights reserved.
+//
+
+import Foundation
+
+class Store {
+    
+    
+    static let shared = Store()
+    private(set) var entries = [MovieReview]()
+    
+    
+    
+    ///creates a new instance of MovieReview, and adds it to the entries array
+    func addEntryWith(title: String, review: String) {
+        let entry = MovieReview(title: title, review: review)
+        entries.append(entry)
+    }
+    
+    
+    
+    ///removes the entry from the entries array
+    func remove(entry: MovieReview) {
+        if let entryIndex = entries.index(of: entry) {
+            entries.remove(at: entryIndex)
+        }
+    }
+    
+    
+    
+    ///takes in an existing entry as a parameter, as well as the title and text strings to update the entry with
+    func update(entry: MovieReview, with title: String, review: String) {
```

Comment:
It would be better for this function to create a new value rather than mutating an existing one. Also, since `MovieReview` is a `struct`, this won't replace the value in the array but instead just mutate the copy in the function. You'd never see the result of this update. I think I remember you talking on Friday about adding a `UUID`, which would be right approach.

Reply:
I added a UUID property to the MovieReview struct, so when I create a MovieReview, it would generate it’s UUID.

`let id = UUID()`

I modified the update function in the Store class so that it is going to iterate through each of the entries and see if it could find an entry that is equal to ‘x’ entry, and it's going to do that by comparing ids.

Also, the array is being accessed directly:
`self.entries[index].title = title`
…so this will update an entry in the array of entries.

Please note that I am not sure if it is overkill with checking id equality, but I can easily take it out:

`static func ==(lhs: MovieReview, rhs: MovieReview) -> Bool {
        return lhs.id == rhs.id
    }`